### PR TITLE
PLAT-114153: Fix Scroller and VirtualList to re-render when its size changed

### PR DESCRIPTION
**Pull request by MikyungKim**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [ ] Documentation was added or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
When a Scroller or VirtualList changed its size dynamically by other elements, for example, in Panel from header height changing, it is not re-rendered so focus can be truncated.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Added `ResizeObserver` to make Scroller or VirtualList re-render when its size has been changed.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
I left `resize` handler as is because it has an additional feature like scrollTo 0 position logic.

### Links
[//]: # (Related issues, references)
PLAT-114153

### CODE REVIEW

File: packages/ui/useScroll/useScroll.js [hunk 1]
```diff
@@ -289,6 +290,7 @@ const useScrollBase = (props) => {
 
 	useEffect(() => {
 		mutableRef.current.resizeRegistry.parent = context;
+		mutableRef.current.resizeObserver.observe(scrollContainerRef.current);
```
Comment (ybsung) [inline packages/ui/useScroll/useScroll.js:293]: Don't we need unobserve() when unmounting?

Comment (MikyungKim) [reply packages/ui/useScroll/useScroll.js:293]: good point!

File: packages/ui/useScroll/useScroll.js [hunk 2]
```diff
@@ -287,6 +288,29 @@ const useScrollBase = (props) => {
...
+			return;
+		}
+		if (typeof ResizeObserver === 'function') {
+			let resizeObserver = new ResizeObserver(() => {
+				if (scrollContentHandle.current && scrollContentHandle.current.syncClientSize) {
+					scrollContentHandle.current.syncClientSize();
+				}
+			});
+
+			resizeObserver.observe(containerRef);
```
Comment (jeonghee27) [inline packages/ui/useScroll/useScroll.js:303]: No need to do resizeObserver.unobserve()?

Comment (MikyungKim) [reply packages/ui/useScroll/useScroll.js:303]: I did disconnect().

Comment (jeonghee27) [reply packages/ui/useScroll/useScroll.js:303]: Ah, I saw resizeObserver.disconnect()

**Pull request merged**